usaspending_api/awards/v2/views/award_spending.py

```diff
@@ -61,39 +56,41 @@ def get_queryset(self):
         # retrieve post request payload
         json_request = self.request.query_params
 
-        # retrieve fiscal_year & awarding_agency_id from request
+        # Retrieve fiscal_year & awarding_agency_id from Request
         fiscal_year = json_request.get('fiscal_year', None)
         awarding_agency_id = json_request.get('awarding_agency_id', None)
 
-        # required query parameters were not provided
+        # Optional Award Category
+        award_category = json_request.get('award_category')
+
+        # Required Query Parameters were not Provided
         if not (fiscal_year and awarding_agency_id):
             raise InvalidParameterException(
                 'Missing one or more required query parameters: fiscal_year, awarding_agency_id'
             )
-
         if not check_valid_toptier_agency(awarding_agency_id):
             raise InvalidParameterException('Awarding Agency ID provided must correspond to a toptier agency')
 
-        # change user provided PK (awarding_agency_id) to toptier_agency_id,
-        # filter and include all subtier_agency_id(s).
         top_tier_agency_id = Agency.objects.filter(id=awarding_agency_id).first().toptier_agency_id
         queryset = Transaction.objects.all()
-        # Filter based on fiscal year and agency id
+
+        # Filter based on fiscal_year, awarding_agency_id, and category
         queryset = queryset.filter(
             fiscal_year=fiscal_year,
-            awarding_agency__toptier_agency=top_tier_agency_id
-        )
-        # alias recipient column names with top and sub tier agencies
-        queryset = queryset.annotate(
-            recipient_name=F('recipient__recipient_name'),
-            recipient_id=F('recipient__legal_entity_id')
+            awarding_agency__toptier_agency=top_tier_agency_id,
+            award__category=award_category
```

Comment:
If `award_category` is not supplied, won't this filter out everything with a non-null `award__category`?  Is that the intent?